examples/tutorials/physics_unit_tests.py

```diff
@@ -0,0 +1,312 @@
+    sim.set_translation(np.array([2.50, 0.35, 1.35]), bowlIDs[1])
+    sim.set_translation(np.array([2.50, 0.35, -0.55]), bowlIDs[2])
+    sim.set_translation(np.array([3.45, 0.35, 0.4]), bowlIDs[3])
+    sim.set_translation(np.array([1.55, 0.35, 0.4]), bowlIDs[4])
+
+    # load some object templates from configuration files
+    object_ids = []
+    scales = {
+        "test_assets/objects/sphere": 1,
+        "test_assets/objects/chair": 0.9,
+        "test_assets/objects/donut": 2,
+        "test_assets/objects/nested_box": 0.5,
+    }
+    for obj_path in objects:
+        object_ids += [
+            obj_templates_mgr.load_configs(str(os.path.join(data_path, obj_path)))[0]
+        ]
+        obj_template = obj_templates_mgr.get_template_by_ID(object_ids[-1])
+        obj_template.scale *= object_scale
+        if obj_path in scales.keys():
+            obj_template.scale *= scales[obj_path]
+        obj_templates_mgr.register_template(obj_template)
+
+    observations = []
+    physics_step_times = []
+    graphics_render_times = []
+    collisions = []
+
+    # throw objects into bowl
+    for i in range(num_objects):
+        cur_id = sim.add_object(object_ids[i % len(object_ids)])
+
+        obj_node = sim.get_object_scene_node(cur_id)
+        obj_bb = obj_node.cumulative_bb
+        diaganal_length = math.sqrt(
+            obj_bb.size_x() ** 2 + obj_bb.size_y() ** 2 + obj_bb.size_z() ** 2
+        )
+        time_til_next_obj = diaganal_length / object_speed / 2
+        # obj_node.scale(mn.Vector3(3,3,3))
+
+        # set object position and velocity
+        sim.set_translation(np.array([1.50, 2, 1.2]), cur_id)
+        initial_linear_velocity = mn.Vector3(1, 0, -1)
+        initial_linear_velocity = initial_linear_velocity.normalized() * object_speed
+        sim.set_linear_velocity(initial_linear_velocity, cur_id)
+
+        # simulate half a second, then add next object
+        (
+            cur_observations,
+            cur_physics_step_times,
+            cur_graphics_render_times,
+            cur_collisions,
+        ) = simulate(sim, dt=time_til_next_obj, get_frames=make_video)
+        observations += cur_observations
+        physics_step_times += cur_physics_step_times
+        graphics_render_times += cur_graphics_render_times
+        collisions += cur_collisions
+
+    (
+        cur_observations,
+        cur_physics_step_times,
+        cur_graphics_render_times,
+        cur_collisions,
+    ) = simulate(sim, dt=post_throw_settling_time, get_frames=make_video)
+    observations += cur_observations
+    physics_step_times += cur_physics_step_times
+    graphics_render_times += cur_graphics_render_times
+    collisions += cur_collisions
+
+    # [/basics]
+    # return total time to run, time to load, time to simulate physics, time for rendering
+    remove_all_objects(sim)
+    sim.close()
+    return physics_step_times, graphics_render_times, collisions, observations
+
+
+unit_tests = {
```

Comment:
I wonder if this structure would benefit from an extra entry with a short description of what a scenario is trying to probe? This could also be recorded in the results log mentioned below. Would make the final output more interpretable.

E.g. "Filling a primitive box vessel with primitive shapes" and "Filling a mesh vessel with convex hull shapes", "Benchmarking dense contact between a variety of convex hull shapes", "Benchmarking raycast performance in a scene", "measuring time-to-sleep in a simple drop scenario", etc...

Reply:
@aclegg3 Any reason why aren't using the pytest-benchmarking plugin for this?